src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -28,19 +28,37 @@ public GpioController(PinNumberingScheme numberingScheme)
         /// Attempt to get the best applicable driver for the board the program is executing on.
         /// </summary>
         /// <returns>A driver that works with the board the program is executing on.</returns>
-        private static GpioDriver GetBestDriverForBoard()
+        public static GpioDriver GetBestDriverForBoard()
         {
             string[] cpuInfoLines = File.ReadAllLines(CpuInfoPath);
-            Regex regex = new Regex(@"Hardware\s*:\s*(.*)");
-            foreach (string cpuInfoLine in cpuInfoLines)
+            Regex regexHw = new Regex(@"Hardware\s*:\s*(.*)");
+            Regex regexRev = new Regex(@"Revision\s*:\s*(.*)");
+            for (int i = 0; i < cpuInfoLines.Length; i++)
             {
-                Match match = regex.Match(cpuInfoLine);
+                string cpuInfoLine = cpuInfoLines[i];
+                Match match = regexHw.Match(cpuInfoLine);
                 if (match.Success)
                 {
                     if (match.Groups.Count > 1)
                     {
                         if (match.Groups[1].Value == RaspberryPiHardware)
                         {
+                            if (i + 1 < cpuInfoLines.Length)
+                            {
+                                string revisionLine = cpuInfoLines[i + 1];
+                                match = regexRev.Match(revisionLine);
+                                if (match.Success && match.Groups.Count > 1)
+                                {
+                                    string rev = match.Groups[1].Value;
+                                    if (rev.EndsWith("03111"))
+                                    {
+                                        // The Pi4 reports the same CPU (although that's technically wrong) but different board revisions.
+                                        // The currently known revisions are 0xa03111, 0xb0311, 0xc03111 for the models with 1GB, 2GB and 4GB of ram, respectively. 
```

Comment:
Do you mean this won't work for 2GB or should this be `0xb03111`? Are the revisions supposed to be the same here?

Reply:
That was a simple typo in the comment. I don't understand your second question.

Reply:
@pgrawehr I didn't notice `a`, `b` and `c` changing and thought those are three the same numbers